rfc/001-identifier-triplets.md

```diff
@@ -0,0 +1,201 @@
+for comparing different fields.  The general process though has been to compare a list of string values for each
+field.  While many fields only have 1 value (eg `SEX`, `RACE`, `BIRTH_DATE`), some fields have multiple values
+(eg `ADDRESS`, `FIRST_NAME`, `LAST_NAME`).  Thus it was important to implement the evaluation functions to
+accept a list of strings for both the incoming document and the existing document when comparing these fields.
+The important thing to note here, is if we are comparing a field like `IDENTIFIER`, we are likely limited to
+just comparing 1 string value per identifier listed in a document (we are assuming more than 1 identifier can
+be listed in a document).
+
+## Proposal
+
+The proposal is to use a 3-tuple of type, authority and value to specify patient identifiers and evaluate them.
+
+- Input: A list of identifier objects, each with a type, authority and value.
+    example: `[{type: 'DL', authority: 'CA', value: 'A123456'}, {type: 'SS', authority: '', value: '123-45-6789'}]`
+- Business Rules: Algorithm configuration will accept the `IDENTIFIER` field with a type suffix, to specify
+    the type of identifier to compare.
+    example: `IDENTIFIER:DL`, `IDENTIFIER:SS`
+- Blocking: The blocking key will be `IDENTIFIER` and values will be inserted for every identifier specified in
+    the document using the type and the last four of the value.
+    example: `DL:CA:3456`, `SS::6789`
+- Evaluation: The evaluation function will accept a list of identifier strings, containing all 3 parts of the
+    tuple, and compare them using the specified evaluation function.
+    example: `['DL:CA:A123456', 'SS::123-45-6789']`
+
+### Input
+
+The input to the Record Linker process will be a list of identifier objects, each with a type, authority and value.
+The type and value attributes are required, while authority is optional.
+
+```json
+{
+    "identifiers": [
+        {
+            "type": "DL",
+            "authority": "CA",
+            "value": "A123456"
+        },
+        {
+            "type": "SS",
+            "authority": "",
+            "value": "123-45-6789"
+        }
+    ]
+}
+```
+
+The type attribute will be limited to a codes defined by the
+[HL7 identifierType code system](https://terminology.hl7.org/6.0.2/CodeSystem-v2-0203.html). This 
+includes roughly 100 different types of identifiers that are all coded using a 2-7 character value.
+For example, `DL` is the code for Driver's License, `SS` is the code for Social Security Number, `MR`
+is the code for Medical Record Number.
+
+The authority attribute will be a free-form string that can be used to specify the issuing authority
+of the identifier.  For example, `CA` could be used to specify that the Driver's License was issued
+by the state of California.  There is no standard code system for this attribute, so it will be up to
+the user to specify a value that makes sense for their data.
```

Comment:
If using the `/link` API yes, if using the `/fhir` API then it will be in the identifying value in the [assigner](https://www.hl7.org/fhir/datatypes-definitions.html#Identifier.assigner).